src/iconv.jl

```diff
@@ -102,11 +108,11 @@ function iconv_reset!(s::Union{StringEncoder, StringDecoder})
                 (Ptr{Void}, Ptr{Ptr{UInt8}}, Ref{Csize_t}, Ptr{Ptr{UInt8}}, Ref{Csize_t}),
                 s.cd, C_NULL, C_NULL, s.outbufptr, s.outbytesleft)
 
-    if ret == reinterpret(Csize_t, -1)
+    if ret == typemax(Csize_t)
```

Comment:
That's not the problem.  You *will* get back a Csize_t value of 0xffffffffffffffff from C.
The problem is that the reinterpret(Csize_t, -1) won't work, not on a 32-bit system, because the -1 will be an Int32, not an Int64.